### DISCUSSION TITLE
Multilingual dashboard and email reports

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#354] [Anything Else]
"Is there a plan to offer the reports in other languages? I could then use the monthly email reports with my clients"

(request via email)

**Comment by mirkoschubert** [comment]
Sorry, I didn't see that one. So I repeat my offer here: I would volunteer to translate from English to German once the i18n implementation is ready. 😄

**Reply by mfru** [reply]
I second that, would be glad to help out by translating!

**Comment by jtraulle** [comment]
I would be also interested, I can translate to French 🇫🇷

**Reply by bmenant** [reply]
I can give a hand over here! ;-)

**Reply by ccgee** [reply]
Présent également ! ;-)

**Comment by uhlhosting** [comment]
I am too intrested, could help with Romanian language.

**Comment by metmarkosaric** [comment]
Thank you! So nice to see that you all are happy to help us with this! Would be great to make Plausible multilingual! 

Our development priority at the moment is on the public API and after that it's probably going to be the import / export. We're a small team so need to prioritize according to the most requested features. 

But let's see, if someone could help contribute by building the internationalization implementation for us it would certainly help speed it up.

**Comment by hejjagheterjonas** [comment]
Yes, i18n would be great! I can help out with Swedish translation.

**Comment by clement-escolano** [comment]
Hello,
I have never used Elixir before but would like to help in internationalisation.
Do you think this [library](https://github.com/elixir-gettext/gettext) is the right tool?
If so, I will try to make a PR and start the miltilingual support :-)

**Comment by ConsultorML** [comment]
I'm very interested, I can translate it into Spanish.

**Comment by rmens** [comment]
Would be able to provide Dutch translations. As a long time user of the open source version of Plausible for a non-profit I would be happy to contribute some time.

**Comment by maximeMD** [comment]
I also think this could be a great improvement, because this tools is aimed to be used (at least through the email reports) by end-users, and as a website developer this could be considered as non "professional" to provide a tool for my clients which is not is their language.

I would be glad to contribute to the translations, but I don't know anything about Elixir/Phoenix/Gettext so I'm not able to implement the i18n system...

Thanks anyway for your amazing work 😉

**Comment by leopeltola** [comment]
I'm also very interested in this feature. Just translating the dashboard and email reports would be great.

I would also gladly help with the Finnish translation.

**Comment by dittaeva** [comment]
As has been said, great product (I am a customer) :-)

Multilingual support is important! Especially in Europe. Not having it, makes you look like an american company (the negative aspects of being one). You should probably use a translation platform from the get-go. [Weblate](https://weblate.org) seems to be among the better and more libre choices. I could contribute to Norwegian translations.

**Comment by SemaphoreOxalis** [comment]
We completely +1 on this one for the saas edition. As a French company, we start to present Plausible to our Matomo/GA users and the ONLY aspect some of them don't like : the English labels.

**Comment by hemju** [comment]
👍  on the multilingual dashboard. Obviously I am based because I am one of the founders of Lingohub. If you guys are interested, I can show you how LH works and maybe it is a good fit. LH is free for open source tools

**Reply by metmarkosaric** [reply]
thanks @hemju! all the requests about localization have been heard and it's definitely something we'd like to support in the future. we will explore Lingohub too. we have limited development resources so need to take things one at a time. currently, our focus is on finishing the GA data import which is a huge undertaking and we plan to look at some ecommerce features after that.

**Reply by hemju** [reply]
Sounds good, let me know if I can help you after the GA.

**Comment by sschueller** [comment]
I looked into this in a bit more detail. The approach I would take is via GetText and set_locale (https://phrase.com/blog/posts/i18n-for-phoenix-applications-with-gettext/). This uses industry standard ```.pot``` an ```.po``` files for translations. Something like https://weblate.org/en/ (Free for open source projects) or alternatives can be used to translate strings. Unit Tests that compare strings will still work as there is a ```default.pot``` which you do not change other than adding new text after first extraction.

However there are currently some issues:

- Using a URL prefix my be undesired especially for the API, the current version of set_locale (https://github.com/smeevil/set_locale) does not support turning this off. There is a fork at (https://github.com/sraka1/set_locale) that supports APIs. set_local also uses language header, cookie and a fallback. It would be desirable if the plugin enabled setting which of these should be used.
- set_locale does not support nested routes, would require duplication in ```router.ex```
- It appears (didn't spend too much time on this) hard-coded routes also need to be adjusted like POST ```/login```.
- set_locale appears to be abandoned :(
- Once a system is in place there will be a not small amount of busy work replacing all string that need translation to use GetText.

My recommendation is to fork set_local and add the stuff you need.

**Reply by ruslandoga** [reply]
👋

I would also suggest using something like CLDR in addition to gettext. Since then you'd also get numbers / dates / units / etc. l10n "for free". [ex_cldr](https://github.com/elixir-cldr/cldr) also has [helpers](https://hexdocs.pm/ex_cldr_plugs/readme.html) that make set_locale unnecessary.

**Comment by fhfa** [comment]
Status update on this feature request?
I volunteer to provide translations to Portuguese of Portugal (pt-PT) 🇵🇹.

**Comment by click0** [comment]
Up.
I'm ready to translate into Ukrainian.

**Comment by uhlhosting** [comment]
Guys move on to some other app, best advice, not many times you see real people trying to make your app better, and the other side sleeps. let them translate it with AI, will be good enough.

**Reply by rmens** [reply]
What a rude comment. You can see in the commits that the Plausible team is working on the application daily. I’m sure this will get planned when the time is right. No need to move on.

**Reply by SemaphoreOxalis** [reply]
I react asyou at first and now I'm quite sure he was talking about Language app as Lingohub, community translators etc

**Comment by MichaelCharles** [comment]
Any update on whether or not this is on the roadmap at all?

**Comment by pipozzz** [comment]
I can help with Slovak
